core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -378,7 +377,7 @@ public final T get(Settings primary, Settings secondary) {
             return get(primary);
         }
         if (fallbackSetting == null) {
-            return get(secondary);
+            return exists(secondary) ? get(secondary) : get(primary);
```

Comment:
I think this block and the following one could be reordered to get the same effect?
```
if (exists(secondary)) {
    return get(secondary);
}
if (fallbackSetting == null) {
    return get(primary);
}